### PR TITLE
EZP-31590: Added update feature for URL Wildcards

### PR DESCRIPTION
**Pull request by michal-myszka**
| Question                                  | Answer
| ---------------------------------------- | ------------------
| **JIRA issue**                          | https://jira.ez.no/browse/EZP-31590
| **Type**                                   | feature
| **Target eZ Platform version** | `v3.1`
| **BC breaks**                          | no
| **Tests pass**                          | yes
| **Doc needed**                       | yes

#### Checklist:
- [x] PR description is updated.
- [x] Tests are implemented.
- [x] Added code follows Coding Standards (use `$ composer fix-cs`).
- [x] PR is ready for a review.

### CODE REVIEW

File: eZ/Publish/API/Repository/URLWildcardService.php [hunk 1]
```diff
@@ -35,6 +36,24 @@ interface URLWildcardService
...
+     *
+     * @throws \eZ\Publish\API\Repository\Exceptions\BadStateException
+     * @throws \eZ\Publish\API\Repository\Exceptions\InvalidArgumentException if the $sourceUrl pattern already exists
+     * @throws \eZ\Publish\API\Repository\Exceptions\UnauthorizedException if the user is not allowed to update url wildcards
+     * @throws \eZ\Publish\API\Repository\Exceptions\ContentValidationException if the number of "*" patterns in $sourceUrl and
+     *          the number of {\d} placeholders in $destinationUrl doesn't match or
+     *          if the placeholders aren't a valid number sequence({1}/{2}/{3}), starting with 1.
+     *
+     * @param URLWildcard $URLWildcard
+     * @param URLWildcardUpdateStruct $updateStruct
```
Comment (adamwojs) [inline eZ/Publish/API/Repository/URLWildcardService.php:50]: nitpick: please use FQCN

File: eZ/Publish/API/Repository/URLWildcardService.php [hunk 2]
```diff
@@ -35,6 +36,24 @@ interface URLWildcardService
...
+     */
+    public function update(
+        URLWildcard $URLWildcard,
+        URLWildcardUpdateStruct $updateStruct
+    ): void;
```
Comment (adamwojs) [inline eZ/Publish/API/Repository/URLWildcardService.php:55]: We should introduce related events to be consistent

File: eZ/Publish/Core/Repository/URLWildcardService.php [hunk 3]
```diff
@@ -115,16 +108,46 @@ public function create(string $sourceUrl, string $destinationUrl, bool $forward
         return $this->buildUrlWildcardDomainObject($spiUrlWildcard);
     }
 
+    public function update(
+        URLWildcard $URLWildcard,
```
Comment (adamwojs) [inline eZ/Publish/Core/Repository/URLWildcardService.php:112]:
```suggestion
        URLWildcard $urlWildcard,
```

File: eZ/Publish/API/Repository/Events/URLWildcard/UpdateEvent.php [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+    private $URLWildcard;
+
+    /** @var \eZ\Publish\API\Repository\Values\Content\URLWildcardUpdateStruct */
+    private $updateStruct;
+
+    public function __construct(
+        URLWildcard $urlWildcard,
+        URLWildcardUpdateStruct $updateStruct
+    ) {
+        $this->URLWildcard = $urlWildcard;
```
Comment (Nattfarinn) [inline eZ/Publish/API/Repository/Events/URLWildcard/UpdateEvent.php:27]: Related to @adamwojs comment, please unify the naming (make it `$urlWildcard*`) as here it is obvious inconsistency. :)

Comment (michal-myszka) [reply eZ/Publish/API/Repository/Events/URLWildcard/UpdateEvent.php]: Changed

**Pull request merged**